CONTRIBUTING.md

```diff
@@ -1,8 +1,117 @@
+
+- PHP 5.6: `datadog/docker-library:ddtrace_php_5_6`
+- PHP 7.0: `datadog/docker-library:ddtrace_php_7_0`
+- PHP 7.1: `datadog/docker-library:ddtrace_php_7_1`
+- PHP 7.2: `datadog/docker-library:ddtrace_php_7_2`
+
+To access a `bash` from the PHP 7.2 container, run the following:
+
+```bash
+$ docker-compose run --rm 7.2 bash
+```
+
+Once inside the container, install the dependencies with Composer.
+
+```bash
+$ composer install
+```
+
+Then install the `ddtrace` extension.
+
+```bash
+$ composer install-ext
+```
+
+> **Note:** You'll need to run the above `install-ext` command to install the `ddtrace` extension every time you access the container's bash for the first time.
+
+You can check that the extension was installed properly.
+
+```bash
+$ php --ri=ddtrace
+```
+
+You should see output similar to the following:
+
+```
+ddtrace
+
+
+Datadog PHP tracer extension
+For help, check out the documentation at https://github.com/DataDog/dd-trace-php/blob/master/README.md#getting-started
+(c) Datadog 2018
+
+Datadog tracing support => enabled
+Version => 0.7.0-beta
+```
+
+When you're done with development, you can stop and remove the containers with the following:
+
+```bash
+$ docker-compose down
+```
+
+### Running the tests
+
+In order to run all the tracer tests:
+
+```bash
+$ composer test
+```
+
+> **Note:** The `composer test` command is a wrapper around `phpunit`, so you can use all the common [options](https://phpunit.de/manual/5.7/en/textui.html#textui.clioptions) that you would with `phpunit`. However you need prepend the options list with the additional `--` dashes that `composer` requires:
+
+```bash
+# Run a suite and a filter
+$ composer test -- --testsuite=unit --filter=Predis
+```
+
+In order to run the `phpt` tests for the php extension:
+
+```bash
+$ composer test-ext
+```
+
+### PHP linting
+
+The PHP tracer conforms to the [PSR-2 coding style guide](https://www.php-fig.org/psr/psr-2/). The code style is checked with [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) which can be invoked with the following command:
+
+```bash
+$ composer lint
+```
+
+To try to automatically fix the code style, you can run:
+
+```bash
+$ composer fix-lint
+```
+
+### Static Analyzer
+
+The [PHPStan static analyzer](https://github.com/phpstan/phpstan) is part of the build checks when submitting a PR. To ensure your contribution passes the static analyzer, run the following:
+
+```bash
+$ composer static-analyze
+```
+
+> **Note:** The static analyzer only works on PHP 7.1 builds and greater.
+
+## Sending a pull request (PR)
+
+There are a number of checks that are run automatically with [CircleCI](https://circleci.com/gh/DataDog/dd-trace-php/tree/master) when a PR is submitted. To ensure your PHP code changes pass the CircleCI checks, make sure to run all the same checks before submitting a PR.
+
+```bash
+$ composer test && composer lint && composer static-analyze
```

Comment:
👍 
Possible idea for the future - "implement  `composer pre-submit-checks`"

Reply:
Ooh! I like that idea! :)